### ISSUE TITLE
Getting error when trying to compile/assemble under Mac OSX 10.13

### ISSUE THREAD

**Issue by petersieg** [blaknite/gigatron#1]
Hi there.

ruby --version
ruby 2.3.7p456 (2018-03-28 revision 63024) [universal.x86_64-darwin17]

ichs-iMac:src ich$ ruby lines.rb
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require': /Users/ich/Downloads/gigatron-master/lib/asm.rb:285: multiple assignment in conditional (SyntaxError)
	from /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from lines.rb:3:in `<main>'
ichs-iMac:src ich$ 

Thx.

**Comment by blaknite** [comment]
Hi Peter, thanks for reaching out!

I pulled down the code just now and ran it with ruby 3.2 without error. When I wrote this, it was probably with Ruby 2.7 with syntax not supported in ruby 2.3. I tried to test against ruby 2.3 on my machine but ruby 2.3 failed to build. Unfortunately I don't have time to debug why. Sorry!

Perhaps you could try installing ruby 2.7 or later with https://github.com/rbenv/rbenv?

**Comment by petersieg** [comment]
Thx looking into it.
Will try to upgrade ruby.
I saw, that the other programs require ImageMagick - to convert png to whatever needed. This has another huge dependency to install on mac os :-( - will see.
Hope that parsec will get some more time to implement opponents etc.
Anyway many thanks for your work - I like puzzle - a nice tetris like game!
Best, Peter

**Comment by petersieg** [comment]
Hi.

Installing (upgrading) ruby on mac os is not that easy, because it is used already by the os itself with an outdated version.
Following this nice writeup: https://snyk.io/de/blog/how-to-install-ruby-in-mac-os/
I was able to install ruby 3.3.0 - and can confirm, that lines.rb can be successfully compiled now.

Thx, Peter

**Comment by blaknite** [comment]
Great to hear you have it working and that you're enjoying puzzles!

When writing parsec I quickly hit up against performance limitations with rendering the starfield. I think it's possible to optimise it, but I'm not sure it'll quite fast enough to be the game I wanted. It might just have to take another look :D

I see that writeup includes using brew to install rbenv and ruby-build. You should be able to run `brew install imagemagick` to resolve that dependency too.

**Comment by petersieg** [comment]
--
ImageMagick has another huge dependency using brew :-(
However, I installed binary from here: https://imagemagick.org/archive/binaries/ImageMagick-x86_64-apple-darwin20.1.0.tar.gz - this is ONLY for High Sierra!
Also I needed to move ImageMagick full directory to / and change the environment variables accordingly.
See: https://imagemagick.org/script/download.php
And then install all required libs with brew.
It was freetype, libXaw, libXt if I remember right (but you get the missing lib message when assembling any program).
I am now able to use your ruby assembler also on puzzles and parsec :-)

Hope that you find the time and will to implement at least some opponents in parsec ;-)

Thx and all the best,
Peter
